### PR TITLE
Tx file propagation

### PR DESCRIPTION
**Pull request by musitdev**
Add Deploy Tx prover and image file propagation among all nodes.

Does a Run Tx can contain a download file? I'm not sure, it works in this case because of the path. The download manager server its file from the image directory. 

I correct an issue with the Tx propagation. The Tx wasn't processed the same way if it came from RPC or p2p. The correction is hard coded to avoid to add another Mutex.

### CODE REVIEW

File: crates/cli/localkey.pki
Comment (tuommaki) [inline crates/cli/localkey.pki:1]: I'd maybe drop this from the repo & add a `.gitignore` entry for `localkey.pki` or `*.pki`.

File: crates/cli/src/keyfile.rs [hunk 1]
```diff
@@ -21,3 +23,5 @@ pub fn read_key_file(file_path: &PathBuf) -> crate::BoxResult<SecretKey> {
     let key_array = fs::read(file_path)?;
     Ok(SecretKey::parse_slice(&key_array)?)
 }
+
+//pub print_pub_key
```
Comment (tuommaki) [inline crates/cli/src/keyfile.rs:27]:
```suggestion
```
TODO or just a leftover?

File: crates/node/src/cli.rs [hunk 2]
```diff
@@ -44,12 +44,7 @@ pub struct Config {
     )]
     pub node_key_file: PathBuf,
 
-    #[arg(
-        long,
-        long_help = "",
-        env = "GEVULOT_P2P_DISCOVERY_ADDR",
-        default_value = "34.88.251.176:9999"
-    )]
+    #[arg(long, long_help = "", env = "GEVULOT_P2P_DISCOVERY_ADDR")]
```
Comment (tuommaki) [inline crates/node/src/cli.rs:47]: Why drop the default value?

File: crates/node/src/main.rs [hunk 3]
```diff
@@ -296,9 +310,13 @@ async fn run(config: Arc<Config>) -> Result<()> {
     .await?;
 
     tracing::info!("gevulot node started");
-    loop {
-        sleep(Duration::from_secs(1));
+    // loop {
+    //     sleep(Duration::from_secs(1));
+    // }
```
Comment (tuommaki) [inline crates/node/src/main.rs:315]:
```suggestion
```

I guess we can drop this?

File: crates/node/src/main.rs [hunk 4]
```diff
@@ -296,9 +310,13 @@ async fn run(config: Arc<Config>) -> Result<()> {
...
+    if let Err(err) = download_jh.await {
+        tracing::info!("dwn mng error:{err}");
```
Comment (tuommaki) [inline crates/node/src/main.rs:317]:
```suggestion
        tracing::info!("download_manager error:{err}");
```
Less confusing if suddenly confronted with this one :sweat_smile:

File: crates/node/src/networking/download_manager.rs [hunk 5]
```diff
@@ -1,21 +1,104 @@
...
+use std::path::Path;
+use tokio::fs::File;
+use tokio::net::TcpListener;
+use tokio::task::JoinHandle;
+use tokio_util::io::ReaderStream;
 
-use crate::{cli::Config, storage};
+// pub struct DownloadManager {
+//     file_server_jh: JoinHandle<()>,
+// }
```
Comment (tuommaki) [inline crates/node/src/networking/download_manager.rs:19]:
```suggestion
```

Let's drop this. I guess it was dead code anyway :see_no_evil:

File: crates/node/src/networking/download_manager.rs [hunk 6]
```diff
@@ -1,21 +1,104 @@
...
 
-pub struct DownloadManager {
-    database: Arc<storage::Database>,
-    file_storage: Arc<storage::File>,
+// impl DownloadManager {
+//     pub async fn new(config: &Config, bind_addr: SocketAddr) -> Result<Self> {
+//         let file_server_jh = serve_file(config).await?;
+//         Ok(DownloadManager { file_server_jh })
+//     }
+// }
```
Comment (tuommaki) [inline crates/node/src/networking/download_manager.rs:26]:
```suggestion
```

Same here.

File: crates/node/src/networking/mod.rs [hunk 7]
```diff
@@ -1,8 +1,8 @@
-mod download_manager;
+pub mod download_manager;
 mod noise;
 pub mod p2p;
 
-#[allow(unused_imports)]
-pub use download_manager::DownloadManager;
+// #[allow(unused_imports)]
+// pub use download_manager::DownloadManager;
```
Comment (tuommaki) [inline crates/node/src/networking/mod.rs:6]:
```suggestion
```

**Review by tuommaki — COMMENTED** [review]
Check the default value of P2P discovery address. Was that intentional?

Otherwise, minor nitpicks/suggestions; in general LGTM :+1:

File: crates/cli/localkey.pki
Comment (musitdev) [reply crates/cli/localkey.pki:1]: Yes good idea I add it. I didn't see I've added it.

File: crates/node/src/cli.rs (hunk 2 above)
Comment (musitdev) [reply crates/node/src/cli.rs:47]: To manage the case the node start without connecting, sort of primary bootstraper. In this case, you specify nothing and the node didn't try to connect.

**Review by tuommaki — APPROVED** [review]
LGTM :+1:

**Comment by tuommaki** [comment]
Folding all PRs for operator onboarding :slightly_smiling_face:

**Pull request merged**